defi/lending_platform/src/lending_platform.rs

```diff
@@ -0,0 +1,436 @@
+                        );
+                    x.take(amount) // This only occurs if the assertion does not fail
+                },
+                None => {
+                    panic!(
+                        "[LendingPlatform][POOL] No asset of type {} are currently in the liquidity pool",
+                        asset_address
+                    );
+                }
+            };
+
+            // Calculate the XRD value of the asset the user is attempting to borrow
+            // TODO: Pull this data from an oracle - right now assuming all assets have a 1:1 ratio with the price of radix
+            let cost_of_asset_in_terms_of_xrd = Decimal::from(10000)/10000;
+            let borrow_amount_in_terms_of_xrd = amount * cost_of_asset_in_terms_of_xrd;
+
+            // Check if user has enough collateral available for the loan
+            let user_available_collateral = self.calculate_available_collateral(&user); // TODO: Continue here
+            info!(
+                "[LendingPlatform][USER:{}] Available collateral in terms of XRD: {}",
+                user_address,
+                user_available_collateral
+            );
+            assert! (
+                user_available_collateral >= borrow_amount_in_terms_of_xrd,
+                "[LendingPlatform][POOL] User does not have enough collateral. Requested loan with \
+                value of {} XRD but only has {} XRD of available collateral.",
+                borrow_amount_in_terms_of_xrd,
+                user_available_collateral
+            );
+
+            // Update User Borrowed Balance
+            user.increase_borrowed_balance(asset_address, amount);
+            self.users.insert(user_auth.resource_address(), user);
+            info!(
+                "[LendingPlatform][USER:{}] Borrowing {} of {}.",
+                user_address,
+                amount,
+                asset_address
+            );
+
+            let updated_asset_amount = self.assets.get(&asset_address).unwrap().amount();
+            info!(
+                "[LendingPlatform][POOL] New liquidity balance for {} is {}.",
+                asset_address,
+                updated_asset_amount
+            );
+
+            (user_auth, borrowed_asset)
+        }
+
+        pub fn repay_asset(&self, user_auth: BucketRef, asset: Bucket) -> BucketRef {
+            let asset_address = asset.resource_address();
+            let amount = asset.amount();
+
+            let mut user = self.users.get(&user_auth.resource_address()).unwrap();
+            let user_address = user.user_address;
+
+            info!(
+                "[LendingPlatform][USER:{}] Repaying {} of {}.",
+                user_address,
+                amount,
+                asset_address
+            );
+
+            match self.assets.get(&asset_address) {
+                Some(x) =>  {
+                    x.put(asset);
+                    let updated_asset_amount = self.assets.get(&asset_address).unwrap().amount();
```

Comment:
I think you can just use `x.amount()` here

Reply:
Nice catch! Updated